### ISSUE TITLE
Different SSIM between piqa and skimage

### ISSUE THREAD

**Issue by quqixun** [francois-rozet/piqa#32]
### Description

Piqa and skimage give different ssim using save input.

### Reproduce

```python
import torch
import numpy as np
import imageio.v2 as iio

from piqa import SSIM, PSNR
from skimage.metrics import structural_similarity
from skimage.metrics import peak_signal_noise_ratio


if __name__ == '__main__':

    # random input
    r = np.random.RandomState(42)
    image1 = r.rand(256, 256, 3)
    image2 = r.rand(256, 256, 3)

    # images in same shape
    # image1 = iio.imread('image1.png') / 255.0
    # image2 = iio.imread('image2.png') / 255.0

    image1_tensor = torch.from_numpy(image1.transpose(2, 0, 1)[None]).float().contiguous()
    image2_tensor = torch.from_numpy(image2.transpose(2, 0, 1)[None]).float().contiguous()

    sk_psnr = peak_signal_noise_ratio(image1, image2)
    sk_ssim = structural_similarity(
        image1, image2,
        win_size=7, sigma=1.5, multichannel=True,
        data_range=1.0, K1=0.01, K2=0.03
    )

    piqa_psnr = PSNR()(image1_tensor, image2_tensor)
    piqa_ssim = SSIM(
        window_size=7, sigma=1.5, n_channels=3,
        channel_avg=True, value_range=1.0,
        padding=False, k1=0.01, k2=0.03
    )(image1_tensor, image2_tensor)

    print(f'SKImage PSNR: {sk_psnr}')
    print(f'SKImage SSIM: {sk_ssim}')
    print(f'PIQA SSIM:    {piqa_psnr.item()}')
    print(f'PIQA SSIM:    {piqa_ssim.item()}')

  # output
  # SKImage PSNR: 7.778893543723596
  # SKImage SSIM: 0.0045113893857824895
  # PIQA SSIM:    7.77889347076416
  # PIQA SSIM:    0.004785331431776285

```

### Expected behavior

SSIMs provided by piqa and skimage are more different than other metrics.
It is more obvious when using real images.
Could piqa be more consistent with skimage?
Thanks.

### Environment

* piqa version: 1.2.2
* pytorch version: 1.12.0+cu113
* python version: 3.8.13
* numpy version: 1.23.0
* scikit-image version: 0.18.3
* imageio version: 2.19.3
* os: Ubuntu 20.04

**Comment by francois-rozet** [comment]
Hello @quqixun, thank you for the report.

Looking at `skimage`'s implementation, I think you should set `gaussian_weights=True` to match the SSIM paper. Unfortunately, in that case the [`truncate` variable is hardcoded](https://github.com/scikit-image/scikit-image/blob/38b595d60befe3a0b4c0742995b9737200a079c6/skimage/metrics/_structural_similarity.py#L154-L157) to `3.5` which corresponds to a window size of `11`.

With these modifications (`window_size=11` and `gaussian_weights=True`), we have

```
SKImage PSNR: 7.778893543723596
SKImage SSIM: 0.004677651424548697
PIQA SSIM:    7.77889347076416
PIQA SSIM:    0.004723809193819761
```

which is better. The remaining differences are hard to diagnose but could come from the floating-point precision of PyTorch (32-bit) or the padding applied during the gaussian filter in `skimage`'s implementation. Note that PIQA's implementation is consistent with other PyTorch implementations (`kornia`, `piq`, `pytorch-msssim`).

**Comment by quqixun** [comment]
Thanks, @francois-rozet .
I will try as your suggestion.
